Éric Vynckec ad comments (#150)
anima-wg/voucher: draft-ietf-anima-rfc8366bis.md

Should SHA256 need to be replaced, then a new YANG module will be published with a new leaf, obsoleting `pinned-domain-pubk-sha256` and `proximity-registrar-pubk-sha256`.

> In the event that more than one of `pinned-domain-pubk-sha256`, `pinned-domain-pubk` or `pinned-domain-cert` are present in a voucher, then the Pledge SHALL prioritize

Reviewer: It would be much easier to say that the MASA MUST NOT use more than one `pinned-domain-*` and the Pledge MUST NOT use more than one `proximity-registrar-*`.

With this priority rule as proposed, there are still corner cases e.g.:

Example 1

* Pledge uses proximity-registrar-pubk
* MASA sends a voucher with pinned-domain-pubk-sha256 and pinned-domain-cert
* Which one to select by Pledge?

Example 2

* Pledge includes multiple proximity-registrar-*
* MASA sends a voucher with multiple pinned-domain* back
* Which one to select by Pledge?

Also for the proposed rule, the matching between the 3 items proximity-* to 3 items pinned-domain-* would need to be spelled out I think - which is easily avoided by just adding the 'MUST NOT' rules.

Reviewer: The 'MUST NOT' is basically what we wanted to achieve with a CHOICE statement but couldn't do easily.

Author: In protocol, saying X MUST NOT emit Z, leaves one with the question as to what a receiver is supposed to do when that is violated.

> In the event that more than one of `pinned-domain-pubk-sha256`, `pinned-domain-pubk` or `pinned-domain-cert` are present in a voucher, then the Pledge SHALL prioritize

Author: ```suggestion
Creators of the Aritfacts (Pledge, Registrar for Voucher-Request, MASA for Voucher) MUST NOT include more than one of the pinned- or proxmity- leaves.
However, in the event that more than one of `pinned-domain-pubk-sha256`, `pinned-domain-pubk` or `pinned-domain-cert` are present in a voucher or voucher-request, then the reader SHALL prioritize
```

the `proximity` entry which it used in its voucher-request artifact, ignoring the others.

> If the voucher is nonce-less (and thus not the result of a voucher-request), then the Pledge SHALL consider the first of the above attributes that it understands, in the order given above.

Reviewer: this rule is also not needed, if we impose a MUST NOT rule.

{{RFC8995, Section 3}} defined a "voucher-request" Artifact as an augmented Artifact from the "voucher" Artifact originally defined in {{RFC8366}}.
That definition has been moved to this document, and translated from the "yang-data" extension {{RFC8040}} to the "sx:structure" extension {{RFC8791}}.

In the event that more than one of `proximity-domain-pubk-sha256`, `proximity-domain-pubk` or `proximity-domain-cert` are present in a voucher request, then the Registrar and MASA SHALL consider them in the order presented here.

> The presence of more than one of these attributes is legal as it may allow a Pledge to operate in both constrained and non-constrained networks.

Reviewer: I don't see a use case for including more than one! Was this discussed on the list? (Sorry I might be a bit behind with that)

> However, on constrained networks it wastes significant amounts of space, and it is discouraged in those environments.

Author: ```suggestion
```

revocation status check because, for instance, the OCSP Responder ({{RFC5280}}) is
not reachable.

> In online deployments using BRSKI or a variant, where the voucher will be delivered online to the Pledge, so a validity period of less than an hour will be suitable.

Reviewer: ```suggestion
In online deployments using BRSKI or a variant, where the voucher will be delivered online to the Pledge, a voucher validity period of less than an hour will be suitable.
```

Grammar fixed and clarified - not sure if this was intended.

> For the {{PRM}} variant however, the voucher may transported by foot (literally),

Reviewer: Note I'm surprised this PR adds so much detailed information about PRM - shouldn't the PRM spec define the optimal validity time ... ? As well as other PRM-specifics elsewhere.  But ok to add that here if IESG has requested it; maybe there's a good reason to have this info here.

Author: Yeah.  We were asked for reasonable values to go here.

and a voucher created just before the end of the day, before a long weekend, might not be conveyed until four or five days later.
For such a situation, a validity period of seven days would be appropriate.

> Alternatively, a lightweight renewal MUST be done at the beginning of the next work day with a default validity of less than one work day ("8 hours") may be more workable.

Reviewer: I don't follow "Alternatively ... MUST ... may be more workable."

> Alternatively, a lightweight renewal MUST be done at the beginning of the next work day with a default validity of less than one work day ("8 hours") may be more workable.

Author: ```suggestion
Instead of allowing the voucher to remain valid through a longer break, a lightweight renewal MUST be done at the beginning of the next work day with a default validity of less than one work day ("8 hours"). 
```

Implementations should be aware that the signed data is only
protected from external modification; the data is still visible.
This potential disclosure of information doesn't affect security
so much as privacy.

When used with {{RFC8995}}, or {{cBRSKI}} then voucher requests and vouchers are conveyed using TLS {{RFC9846}}, so there is no exposure.

When used with {{PRM}}, then the contents can be exposed in the last hop,
where HTTP is used, due the lack of any way to validate the certificate needed to enable HTTPS.

When the voucher is in CMS format, it can contain certificate chains that can disclose information such as which devices belong to which organizations
and which CRL Distribution Point and/or OCSP Responder URLs are
accessed to validate the Vouchers.

> Note that {{PRM}} specifies use of {{JWS}} format artifacts rather than CMS, so there are no certificates CRLs to disclose.

Reviewer: ```suggestion
Note that {{PRM}} specifies use of {{JWS}} format artifacts rather than CMS, so there are no CRLs to disclose.
```

The "C" in CRL already stands for 'certificate', so the word shouldn't be added in front of "CRLs", I think.
